lib/std/hash/auto_hash.zig

```diff
@@ -94,13 +94,16 @@ pub fn hash(hasher: anytype, key: anytype, comptime strat: HashStrategy) void {
         .Type,
         .EnumLiteral,
         .Frame,
-        .Float,
         => @compileError("cannot hash this type"),
 
         // Help the optimizer see that hashing an int is easy by inlining!
         // TODO Check if the situation is better after #561 is resolved.
         .Int => @call(.{ .modifier = .always_inline }, hasher.update, .{std.mem.asBytes(&key)}),
 
+        .Float => |info| {
+            var norm_key = if (key == 0 and std.math.signbit(key)) 0.0 else key;
```

Comment:
Is the `signbit` necessary ? Since 0.0 and -0.0 both compare equal to 0, could we just check the equality and then force use of `0` ?

Reply:
If you want to convert `-0` to `0`, you can just add `0`. No need for checks or conditionals.